SwiftAppTemplate/SwiftAppTemplate/MainViewController.swift

```diff
@@ -0,0 +1,171 @@
+            paymentTextField.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -MainViewController.spacing),
+            paymentTextField.heightAnchor.constraint(equalTo: connectReaderButton.heightAnchor),
+        ])
+
+
+        collectPaymentButton.translatesAutoresizingMaskIntoConstraints = false;
+        NSLayoutConstraint.activate([
+            collectPaymentButton.topAnchor.constraint(equalTo: paymentTextField.bottomAnchor, constant: MainViewController.spacing),
+            collectPaymentButton.centerXAnchor.constraint(equalTo: containerView.centerXAnchor),
+            collectPaymentButton.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+            collectPaymentButton.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: MainViewController.spacing),
+            collectPaymentButton.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -MainViewController.spacing),
+        ])
+
+        loadingContainerView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingContainerView.topAnchor.constraint(equalTo: view.topAnchor),
+            loadingContainerView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            loadingContainerView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            loadingContainerView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+        ])
+
+        loadingIndicator.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingIndicator.centerXAnchor.constraint(equalTo: loadingContainerView.centerXAnchor),
+            loadingIndicator.centerYAnchor.constraint(equalTo: loadingContainerView.centerYAnchor),
+        ])
+
+        loadingLabel.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingLabel.topAnchor.constraint(equalTo: loadingIndicator.bottomAnchor, constant: MainViewController.spacing),
+            loadingLabel.centerXAnchor.constraint(equalTo: loadingContainerView.centerXAnchor)
+        ])
+    }
+
+
+    @objc func discoverReader(sender: UIButton) {
+        loadingLabel.text = "Discovering Readers..."
+        loadingContainerView.isHidden = false
+        loadingIndicator.startAnimating()
+
+        // TODO once you have discovered some readers create a
+        // ReaderListViewController and push it onto the navigationController.
+
+        // let readerViewController = ReaderListViewController(readers)
+        // navigationController?.pushViewController(readerViewController, animated: true)
+    }
+
+    @objc func collectPayment(sender: UIButton) {
+        startCollectingPayment()
+    }
+
+    // Mark: - UITextFieldDelegate
+
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
```

Comment:
nit: I wonder if we should remove this, just for simplicity? (might be a bit surprising that Enter triggers payment)

Reply:
The only reason I added this was incase the collectpayment button covered an iPhone 5s. I could put a UIScrollView instead though.